packages/asset-swapper/src/swap_quoter.ts

```diff
@@ -523,26 +530,48 @@ export class SwapQuoter {
             gasPrice = await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync();
         }
         // get the relevant orders for the makerAsset
-        let prunedOrders = await this._getSignedOrdersAsync(makerAssetData, takerAssetData);
+        const orderPromises: Array<Promise<SignedOrder[]>> = [
+            this._getSignedOrdersAsync(makerAssetData, takerAssetData),
+        ];
+        if (options.intentOnFilling && options.apiKey) {
+            if (!this.rfqtTakerApiKeyWhitelist.includes(options.apiKey)) {
+                throw new Error('API key not permissioned for RFQ-T');
```

Comment:
Is this a throwable validation at this layer? Or should it fail silently and give a normal quote? 

If someone gets delisted they're going to start getting Errors rather than quotes, seems quite disruptive.

Reply:
Yeah, agree that we should fallback to providing a regular quote in this case.

Reply:
Addressed in https://github.com/0xProject/0x-monorepo/pull/2541/commits/5f4778c852ed3d2dbed8db04d92f0786343e4c78